test.sh.in

```diff
@@ -252,12 +252,33 @@ nodesCreate() {
   fi
 }
 
+vagrantSuspendAll() {
+ for box in `vagrant global-status --prune | awk '/running/{print $1}'`; do
+     vagrant suspend ${box}
+ done
+}
+
 CONFIG_PATH=${CONFIG_PATH-build-config/vagrant/config/mongo}
 vagrantUp() {
   cd ${WORKSPACE}/RackHD/example
   cp -rf ${WORKSPACE}/build-config/vagrant/* .
-  CONFIG_DIR=${CONFIG_PATH} WORKSPACE=${WORKSPACE} REPO_NAME="${REPO_NAME}" MULTI=${MULTI} vagrant up --provision
-  if [ $? -ne 0 ]; then 
+  result=0
+  for i in {1..2}
+  do
+      CONFIG_DIR=${CONFIG_PATH} WORKSPACE=${WORKSPACE} REPO_NAME="${REPO_NAME}" MULTI=${MULTI} vagrant up --provision
+      result=$?
+      if [ $result -ne 0 ]; then
+          echo "Vagrant up failed!! retry one more time"
+          vagrant destroy -f
+          vagrantSuspendAll
+          rm -rf "$HOME/VirtualBox VMs"
+          rm "$HOME/.config/VirtualBox/VirtualBox.xml"
```

Comment:
1. can you please embed comment in code, to tell the reason you delete the xml file ?
2. I'm still not sure, if we have ```vagrant destroy -f``` running at the end (the ```trap```), why should we still need to remove the xml file ?